lode-ui/src/Model.js

```diff
@@ -1,28 +1,71 @@
-import { OrbitControls, useGLTF } from "@react-three/drei";
-import { Canvas } from "@react-three/fiber";
-import { Suspense, useRef } from "react";
+import { Box, Flex, Spinner } from "@chakra-ui/react";
+import { OrbitControls } from "@react-three/drei";
+import { Html } from "@react-three/drei";
+import { Canvas, useThree } from "@react-three/fiber";
+import { useEffect, useMemo, useState } from "react";
+import { GLTFLoader } from "three/examples/jsm/loaders/GLTFLoader";
 
 function Model({ url }) {
-  const statsParent = useRef(null);
   return (
-    <div ref={statsParent}>
+    <Box>
       <Canvas>
-        <OrbitControls />
+        <OrbitControls autoRotate />
         <ambientLight />
         <pointLight position={[10, 10, 10]} />
 
-        <Suspense fallback={null}>
-          <GltfModel url={url} />
-        </Suspense>
+        <GltfModel url={url} />
       </Canvas>
-    </div>
+    </Box>
+  );
+}
+
+function TriangleCount({ url }) {
+  const { gl } = useThree();
+  const [triangleCount, setTriangleCount] = useState(0);
+  useEffect(() => {
+    setTriangleCount(0);
+    setTimeout(() => {
+      setTriangleCount(gl.info.render.triangles);
+    }, 500);
+    // eslint-disable-next-line
```

Comment:
why?

Reply:
To not trigger the method when the gl object changes.

Reply:
I suggest to store it in a variable outside of the effect and use that variable for the effect change detection then. This way the setTimeout should actually not be required anymore and the whole code works nicer?

Reply:
Then there is no re-render to trigger the effect. Only other option i see would be to use the `useFrame` callback to update it, but seems like an overkill to calculate this every frame, as it is not supposed to change.

Reply:
How do you guarantee that the triangle count is correct after 500 ms?

Reply:
I dont :D I am aware of that.

Reply:
Fair - I don't think we need to over-optimize (as long as it works) 👍 
My only suggestion is to add a comment to answer the initial question "why?". This prevents future me or you of trying to improve it 😄 
something along the lines: "gl.info.render.triangles is a ref, therefore it cannot be used in effect, 500ms is sufficient for 99 percentile, whatever the reasoning"
In general: I try to add comments wherever I assume that someone with sufficient context might ask oneself "why? 🤔 "